brevproduksjon/src/main/java/no/nav/foreldrepenger/fpformidling/brevproduksjon/mapper/felles/SvpMapperUtil.java

```diff
@@ -74,4 +73,29 @@ public static int finnAntallArbeidsgivere(List<SvpUttakResultatArbeidsforhold> u
         }
         return antallArbeidsgivere;
     }
+
+    public static Optional<LocalDate> finnOpphørsdato(List<TilkjentYtelsePeriode> tilkjentYtelsePerioder) {
+        return  tilkjentYtelsePerioder.stream()
+            .filter(tilkjentYtelsePeriode -> tilkjentYtelsePeriode.getDagsats() > 0)
+            .map(TilkjentYtelsePeriode::getPeriodeTom)
+            .max(LocalDate::compareTo)
+            .map(localDate -> justerForHelg(localDate.plusDays(1)));
+    }
+
+    public static LocalDate justerForHelg(LocalDate date) {
```

Comment:
Lite trick: finnes også i LocalDateInterval-klassen - langt nede i koden
private static final Set<DayOfWeek> WEEKEND = Set.of(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY);
return WEEKEND.contains(DayOfWeek.from(date)) ? date.with(next(DayOfWeek.MONDAY)) : date;

Reply:
Forsøkte meg på den i stedet